backend/routes/post.js

```diff
@@ -53,6 +53,29 @@ router.get('/', (request, response, next) => {
 router.get('/:_pid', async (request, response, next) => {
   let body = request.post.toObject()
   const user = await User.getUser(request, response, next, request.post.author)
+  for (let i = 0; i < body.comments.length; i++) {
```

Comment:
These are awesome codes, although I think some way to make it easier to read / better if we follow the modularization pattern of MVC where we put functions that manipulate the model inside the controller, and we use the routes handle calling controllers, and handling errors prior to calling the controller.

 In that case, we separate the logic of data manipulation and interfacing of the backend to the frontend/requester.